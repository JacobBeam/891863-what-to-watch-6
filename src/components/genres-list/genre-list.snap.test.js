import React from 'react';
import {render} from '@testing-library/react';
import {createMemoryHistory} from 'history';
import * as redux from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import {Router} from 'react-router-dom';
import GenreList from './genres-list';


const api = jest.fn();
const middlleware = thunk.withExtraArgument(api);
const mockStore = configureStore([middlleware]);
const setStartCountFilmsInList = jest.fn();

it(`Should GenreList render correctly`, () => {
  const history = createMemoryHistory();
  const store = mockStore({
    FILM_DATA: {films: [],
      genre: `Test`},
  });

  const {container} = render(
      <redux.Provider store={store}>
        <Router history={history}>
          <GenreList setStartCountFilmsInList={setStartCountFilmsInList}/>
        </Router>
      </redux.Provider>
  );
  expect(container).toMatchSnapshot();
});
