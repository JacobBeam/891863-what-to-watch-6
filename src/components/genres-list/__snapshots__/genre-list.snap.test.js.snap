// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should GenreList render correctly 1`] = `
<div>
  <ul
    class="catalog__genres-list"
  >
    <li
      class="catalog__genres-item "
    >
      <a
        class="catalog__genres-link"
        href="#"
      >
        All genres
      </a>
    </li>
  </ul>
</div>
`;
